omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ReportRequestResource.java

```diff
@@ -140,7 +221,10 @@ public DelegatingResourceDescription getRepresentationDescription(Representation
 			description = new DelegatingResourceDescription();
 			description.addProperty("uuid");
 			//description.addProperty("baseCohort", Representation.DEFAULT);  TODO: Figure out how to support this
-			//description.addProperty("reportDefinition", Representation.DEFAULT);  TODO: Figure out how to support this
+			description.addProperty("parameterizable", "reportDefinition.parameterizable",
+					Representation.DEFAULT);
+			description.addProperty("parameterMappings", "reportDefinition.parameterMappings",
+					Representation.DEFAULT);

```

Comment:
Well, both the `reportDefinition` and the `baseCohort` implement `Parameterizable`, so that isn't really a good property name here.  It would be better if the property added was `reportDefinition`, and the value is a representation consisting of `parameterizable`, which would have a `uuid` property, and a `parameterMappings`, which would be a representation of a Map

Reply:
this is probably handled (not by me) in setProperty method so I didn't change it. When property name is reportDefinition then it is mapped to ReportDefinition class, when property name is baseCohort then it is mapped to CohortDefinition class.